pkg/plugins/policies/meshtimeout/plugin/v1alpha1/plugin.go

```diff
@@ -74,11 +74,24 @@ func (p plugin) Apply(rs *core_xds.ResourceSet, ctx xds_context.Context, proxy *
 		}
 	}
 
-	err := applyToRealResources(util_slices.Filter(rs.List(), core_xds.HasResourceOrigin), policies.ToRules.ResourceRules, ctx.Mesh.Resources)
-	if err != nil {
-		return err
-	}
+	rctx := outbound.RootContext[api.Conf](ctx.Mesh.Resource, policies.ToRules.ResourceRules)
+
+	for _, r := range util_slices.Filter(rs.List(), core_xds.HasResourceOrigin) {
+		switch r.ResourceOrigin.ResourceType {
+		case meshservice_api.MeshServiceType:
+		case meshexternalservice_api.MeshExternalServiceType:
+		case meshmultizoneservice_api.MeshMultiZoneServiceType:
+		default:
+			continue
```

Comment:
nitpick: to not duplicate this should we extend the `Filter` to accept multiple conditions and have something like:

```suggestion
	for _, r := range util_slices.Filter(rs.List(), core_xds.HasResourceOrigin, core_xds.IsNotMeshStarService) {
```

and have 

```
func IsNotMeshStarService(r *Resource) bool {
	switch r.ResourceOrigin.ResourceType {
	case meshservice_api.MeshServiceType:
	case meshexternalservice_api.MeshExternalServiceType:
	case meshmultizoneservice_api.MeshMultiZoneServiceType:
		return false
	}
	return false
}
```

```
func Filter[A any](input []A, functions... func(A) bool) []A {
	output := make([]A, 0, len(input))
	for _, a := range input {
		include := true
		for _, f := range functions {
			if ok := f(a); !ok {
				include = false
			}
		}
		if include {
			output = append(output, a)
		}
	}
	return output
}
```

(warning, not tested!)

Reply:
I created filter:
```go
func HasAssociatedServiceResource(r *Resource) bool {
	if r.ResourceOrigin == nil {
		return false
	}
	switch r.ResourceOrigin.ResourceType {
	case
		meshservice_api.MeshServiceType,
		meshexternalservice_api.MeshExternalServiceType,
		meshmultizoneservice_api.MeshMultiZoneServiceType:
		return true
	}
	return false
}
```